src/ui/public/timefilter/time_history.js

```diff
@@ -0,0 +1,85 @@
+import _ from 'lodash';
+import moment from 'moment';
+import { Storage } from 'ui/storage';
+import { TIME_MODES } from 'ui/timepicker/modes';
+
+const MAX_HISTORY = 20;
+const TIME_HISTORY_KEY = 'kibana.timepicker.timeHistory';
+const TIME_KEY = 'kibana.timepicker.time';
+const REFRESH_INTERVAL_KEY = 'kibana.timepicker.refreshInterval';
+
+class TimeHistory {
+  constructor() {
+    this.storage = new Storage(window.localStorage);
```

Comment:
Instead of using the `Storage` class, it might make more sense to use the [`PersistedLog`](https://github.com/elastic/kibana/blob/faf0633232f142e814949dc6839b7b167991de89/src/ui/public/persisted_log/persisted_log.js) class, which already takes care of duplicates.